client/src/Components/Annotation/AnnotationViewer.tsx

```diff
@@ -25,24 +36,51 @@ const AnnotationViewer = (props) => {
     const annot_pos_y = e.target.id().split('-')[2];
     annotations[annot_indx][annot_pos_x] = x / spectrogramWidthScale; // reverse the calcs done to generate the coords
     annotations[annot_indx][annot_pos_y] = y;
-    forceUpdate(); // TODO remove the forceupdate and do it the proper way (possibly using spread?)
-
-    // Now update the actual meta.annotations
-    const f = annotations[annot_indx]['index']; // remember there are 2 different indexes- the ones on the screen and the meta.annotations
-    const updatedAnnotations = [...meta.annotations];
-    const updatedAnnotation = { ...updatedAnnotations[f] };
     const start_sample_index = lowerTile * TILE_SIZE_IN_IQ_SAMPLES;
-    updatedAnnotation['core:sample_start'] = annotations[annot_indx].y1 * fftSize * zoomLevel + start_sample_index;
-    updatedAnnotation['core:sample_count'] =
-      (annotations[annot_indx].y2 - annotations[annot_indx].y1) * fftSize * zoomLevel;
     const lower_freq = meta.captures[0]['core:frequency'] - meta.global['core:sample_rate'] / 2;
-    updatedAnnotation['core:freq_lower_edge'] =
+    const f = annotations[annot_indx]['index'];
```

Comment:
Would it make sense to encapsulate some of these calculations as well? Such as setSampleStart(...), setSampleCount(...) and so on.

Reply:
Probably something we could add to new prs